pkg/operator/workloads/workflow.go

```diff
@@ -295,31 +297,32 @@ func IsDeploymentUpdating(appName string) (bool, error) {
 
 		isSucceeded, err := workload.IsSucceeded(ctx)
 		if err != nil {
-			return false, err
+			return resource.UnknownDeploymentStatus, err
 		}
 		if isSucceeded {
 			continue
 		}
 
 		isFailed, err := workload.IsFailed(ctx)
 		if err != nil {
-			return false, err
+			return resource.UnknownDeploymentStatus, err
 		}
 		if isFailed {
-			continue
+			return resource.ErrorDeploymentStatus, nil
```

Comment:
Reminder to discuss the desired behavior here (it affects the displayed statuses, as well as the situations when the user needs `-f` to replace an updating deployment)